### PR TITLE
Makes whiskey outpost rarer + adds pop lock

### PR DESCRIPTION
**Pull request by ihatethisengine**
# About the pull request
Whiskey outpost is available 3 times less often + requires 140 players to be online.

# Explain why it's good for the game
The gamemode get into the vote way more often than once per 5 day, plus for some reason can be rolled during lowpop hours (and this gamemode is awful on lowpop).

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl: ihatethisengine
add: Whiskey outpost is voteable less often and requires 140 players.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/colonialmarines/whiskey_outpost.dm [hunk 1]
```diff
@@ -75,7 +75,7 @@
 
 	hardcore = TRUE
 
-	votable = TRUE
+	votable = FALSE
 	vote_cycle = 25 // approx. once every 5 days, if it wins the vote
```
Comment (ihatethisengine) [inline code/game/gamemodes/colonialmarines/whiskey_outpost.dm:79]:
```suggestion
	votable = TRUE
	vote_cycle = 75 // approx. once every 5 days, if it wins the vote
```
Alternative solution. It's intended to be available once per 5 days but you can check in discord that sometimes it once per day.

**Comment by hislittlecuzingames** [comment]
APPROVE!

FORCE MERGE!!!!!!

File: code/game/gamemodes/colonialmarines/whiskey_outpost.dm (hunk 1 above)
Comment (Releasethesea) [reply code/game/gamemodes/colonialmarines/whiskey_outpost.dm:79]: whiskey outpost is like zombies, not all people like it but its still fun and a good bit of players do like it :(

Comment (ihatethisengine) [reply code/game/gamemodes/colonialmarines/whiskey_outpost.dm:79]: zombies is not a voteable gamemode. Admins would still be able to force whiskey if they want.

**Comment by BadAtThisGame302** [comment]
Whiskey Outpost is one of the things keeping CM from being stale and it's a change of pace which I personally think is good if it was changed a bit perhaps it'd be perfect.

**Comment by ihatethisengine** [comment]
> Whiskey Outpost is one of the things keeping CM from being stale and it's a change of pace which I personally think is good if it was changed a bit perhaps it'd be perfect.

If I want change of pace I play another game, if I want to play CM I don't want to play Whiskey Outpost

**Comment by ihatethisengine** [comment]
> That's not a removal that's disabling

what tag should I use? Fix? Add? Qol?

**Comment by Googles-Hands** [comment]
> If I want change of pace I play another game, if I want to play CM I don't want to play Whiskey Outpost

Whiskey Outpost is capped at one hour. Go touch some grass, maybe cook a healthy meal if you truly cannot bear spending one hour not playing CM.
Relatedly "Why it's good for the game" has not been properly explained.

**Comment by ihatethisengine** [comment]
> Whiskey Outpost is capped at one hour. Go touch some grass, maybe cook a healthy meal if you truly cannot bear spending one hour not playing CM.

Ok bro go watch the paint dry if you enjoy WO, you'll get roughly the same experience. Some people cannot play the game all day just to wait for next round, and even if they can, why would they NOT play CM if they want play CM? 

> Relatedly "Why it's good for the game" has not been properly explained.

There is nothing left to explain, everyone knows about tons of issues WO has and about the lack of anyone interested in coding for WO.

**Comment by Googles-Hands** [comment]
> Ok bro go watch the paint dry if you enjoy WO, you'll get roughly the same experience. Some people cannot play the game all day just to wait for next round, and even if they can, why would they NOT play CM if they want play CM?

The fact that people vote for it, shows that  a majority of people on the server DO want to play it. If it were that hated, it would never be played, so as it stands it would be like removing Soro because "it's bad, many don't like it and it gets voted once a day at max".

> There is nothing left to explain, everyone knows about tons of issues WO has and about the lack of anyone interested in coding for WO.

That would be a fair point, but as of right now, there is one issue open concerning Whiskey Outpost, if the balance complaints and issue reports were to be of a massive amount I could see how arguing for it's removal would be sensible.
But if the major complaint is that it's just a massive FOB defence, then that is intended and not something to complain about.

**Comment by Nanu308** [comment]
I think the one of the main issues is that WO can be voted on lowpop, it should be population locked as well as only be every X rounds to not become stale. 

I don't necessarily think removing WO is the best choice, I'd rather try the above or other changes.

Besides that, just the "Please" is not a fulfilling reasoning for removing an entire gamemode. Fill that out and give reasoning, until then I'm putting the DNM label on.

**Comment by ihatethisengine** [comment]
Done

**Comment by Googles-Hands** [comment]
> I think the one of the main issues is that WO can be voted on lowpop, it should be population locked as well as only be every X rounds to not become stale.

I brought this up before on the discord. I'm gonna repeat it here incase anyone intends to add the highpop requirement.
So:

In the past Whiskey Outpost, like all other big maps, had a population limit of 130 to be voted on, back when CM was still on gitlab Geeves changed (https://gitlab.com/cmdevs/colonial-warfare/-/merge_requests/2771) how the Whiskey Outpost vote worked.
He changed it from just being a mapvote to the current gamemode vote first, then just a map vote with the single map. He explicitely removed the minimum population limit and stated this in his MR, this was approved by you Nanu, I haven't found any real justification for the removal of the pop-limit alongside the vote change, it's probably somewhere in the discord.

Since I found this strange and 80 player WO was/is bad, I asked around the discord and got approval to re-add the 130 limit. (https://discord.com/channels/150315577943130112/745447048261795890/1044613870070804500)
But I then found out I can't just add the one-line-change pop requirement as it was, since that only prevents maps from being voteable, and with Geeves' MR, WO was changed to be voted as a gamemode first, meaning a 130 limit would only prevent a map from being voted on after WO was already selected as the next rounds gamemode. His gamemode vote couldn't be changed so easily and that was beyond me.
So I asked Geeves if he could try but he said he hadn't looked at the gamemode vote system for a long time. (https://discord.com/channels/150315577943130112/745447048261795890/1045054704369012776)

So Whiskey Outpost has been stuck like this since March of last year, I suppose I have to concede that indeed, no one wants to work on WO.

**Comment by Diegoflores31** [comment]
> So Whiskey Outpost has been stuck like this since March of last year, I suppose I have to concede that indeed, no one wants to work on WO.

whiskey has received some updates thought the year the most recent being one made by Fira like 1 month ago

**Comment by Diegoflores31** [comment]
> You mean the req update? Then you should also count pred removal, that happened just yesterday. Obviously when we are talking about WO development we mean important gameplay changes, not some minor details.

By that logic lets just remove big red and soro because they havent got any major changes in half a year

**Comment by Googles-Hands** [comment]
> whiskey has received some updates thought the year the most recent being one made by Fira like 1 month ago

I was talking about it being stuck without a pop limit.

**Comment by Diegoflores31** [comment]
> > whiskey has received some updates thought the year the most recent being one made by Fira like 1 month ago
> 
> I was talking about it being stuck without a pop limit.

I guess lowpop players want WO too 😈

**Comment by ihatethisengine** [comment]
> By that logic lets just remove big red and soro because they havent got any major changes in half a year

Logic is not your strong side. Big red doesn't have the same amount of problems as WO, Soro is not the best map surely, but still it is at least not a cadehug/suicide simulator either. Why would you compare incomparable?

**Comment by Steelpoint** [comment]
Sometimes you may only have enough time to commit to one round or even half of one round a day, depending on real life issues. Logging on to find the game is set to Whisky Outpost is disheartening. WO is not CM. 

Furthermore, I genuinely think that Whisky Outpost is borderline insulting to xenomorph players. These players are being expected to spawn in as literal canon fodder and are told to rush and die to the Marines for an hour. 

If an admin made a event where half the server were spawned in as Lesser Drones and were made to charge at a marine and die over and over, it'd be called a bad event. But make it a game mode and its perfectly fine. 

I support removing or disabling it.

**Comment by ihatethisengine** [comment]
We need to move back to gitlab

**Review by Zonespace27 — CHANGES_REQUESTED** [review]
Seconding @Nanu308 's first two points.

**Comment by ihatethisengine** [comment]
alright Ill make it more rare and figure pop lock

**Review by Zonespace27 — APPROVED** [review]

**Pull request merged**